### PR TITLE
Add support for array operations to compressed DML

### PR DESCRIPTION
**Pull request by svenklemm**
Depends on #6877

### CODE REVIEW

**Comment by svenklemm** [comment]
Depends on #6877

File: tsl/src/compression/compression.c [hunk 1]
```diff
@@ -2682,13 +2695,63 @@ process_predicates(Chunk *ch, CompressionSettings *settings, List *predicates, L
...
 				}
 			}
 			break;
+			case T_ScalarArrayOpExpr:
+			{
+				ScalarArrayOpExpr *sa_expr = castNode(ScalarArrayOpExpr, node);
+				if (!ts_extract_expr_args(&sa_expr->xpr, &var, &expr, &opno, &opcode))
+					continue;
+
+				if (!IsA(expr, Const))
```
Comment (akuzm) [inline tsl/src/compression/compression.c:2713]: Might be simple to run in once on the entire predicate at the start of this loop, it's recursive.

File: tsl/src/compression/compression.c [hunk 2]
```diff
@@ -2682,13 +2695,63 @@ process_predicates(Chunk *ch, CompressionSettings *settings, List *predicates, L
...
+				{
+					expr = (Expr *) estimate_expression_value(&root, (Node *) expr);
+					if (!IsA(expr, Const))
+						continue;
```
Comment (akuzm) [inline tsl/src/compression/compression.c:2717]: Let's add some test with volatile function just for coverage.

**Review by akuzm — APPROVED** [review]

File: tsl/src/compression/compression.c (hunk 2 above)
Comment (svenklemm) [reply tsl/src/compression/compression.c:2717]: done

File: tsl/src/compression/compression.c (hunk 1 above)
Comment (svenklemm) [reply tsl/src/compression/compression.c:2713]: I'll leave this for another PR, it has the chance of resulting in more work that we can't benefit from, eg constraints on columns we cant filter on for compressed dml

**Pull request merged**